app/models/classification.rb

```diff
@@ -203,14 +201,12 @@ def self.create_category!(options)
   end
 
   def self.categories(region_id = my_region_number, ns = DEFAULT_NAMESPACE)
-    cats = where(:classifications => {:parent_id => 0}).includes(:tag, :children)
-    cats = cats.in_region(region_id) if region_id
+    cats = is_category.in_region(region_id).includes(:tag, :children)
```

Comment:
Does it matter that the `if region_id` conditional was dropped here?

Reply:
Turns out `classifications` is never called with any arguments. So this should be good.

I am curious about the select on namespace. If it is a category, it seems the namespace can only be the default. Getting rid of the `select` (forcing it into ruby vs sql) doesn't sit well with me.

Reply:
I can confirm it's never called with arguments.  Can even remove the args if you wish.

select — as long as `ns` it not part of `classifications` table but is computed by ruby string slicing of associated tag name, I feel it's cleaner to keep it in ruby (rather than JOIN with LIKE query which risks diverging from the ruby definition).
performance-wise, `includes(:tag` makes it ruby select cheap.

> If it is a category, it seems the namespace can only be the default. 

why?  do other tag namespaces use bare Tag without Classification?

Reply:
@cben I have no idea what I wrote. I am even having trouble parsing what I wrote here.

If you `include(:tag)`, it would seem a bunch of `taggings` records would come back. Maybe?

But either way, I agree. Safer to just all as is.